### PR TITLE
validate service parameter in client side to avoid api call in `docker service scale`

### PR DESCRIPTION
**Pull request by allencloud**
**\- What I did**
This PR adds parameter validation in the client side of command `docker service scale ...`

Since in the original implementation, docker cli first fetch service object via api call, then validate parameters. It will add workload in the daemon side and this kind of load is unnecessary.

This PR adds a cli test as well.

Signed-off-by: allencloud allen.sun@daocloud.io

### CODE REVIEW

File: cli/command/service/scale.go [hunk 1]
```diff
@@ -58,12 +65,11 @@ func runScale(dockerCli *command.DockerCli, args []string) error {
 	return fmt.Errorf(strings.Join(errors, "\n"))
 }
 
-func runServiceScale(dockerCli *command.DockerCli, serviceID string, scale string) error {
+func runServiceScale(dockerCli *command.DockerCli, serviceID string, uintScale uint64) error {
```
Comment (thaJeztah) [inline cli/command/service/scale.go]: I don't think the variable needs to be renamed to include the type, just `scale` should do

File: cli/command/service/scale.go [hunk 2]
```diff
@@ -72,17 +78,14 @@ func runServiceScale(dockerCli *command.DockerCli, serviceID string, scale strin
 	if serviceMode.Replicated == nil {
 		return fmt.Errorf("scale can only be used with replicated mode")
 	}
-	uintScale, err := strconv.ParseUint(scale, 10, 64)
-	if err != nil {
-		return fmt.Errorf("invalid replicas value %s: %s", scale, err.Error())
-	}
+
```
Comment (thaJeztah) [inline cli/command/service/scale.go]: Thinking if simply moving this check "up" inside the function, instead of moving it outside would be cleaner (open for input from other maintainers here)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left some suggestions / thoughts

File: cli/command/service/scale.go (hunk 1 above)
Comment (allencloud) [inline cli/command/service/scale.go]: Updated. Thanks @thaJeztah

File: cli/command/service/scale.go (hunk 2 above)
Comment (allencloud) [inline cli/command/service/scale.go]: I was thinking `runServiceScale` had better do some specific things. Validation is not related to scale. Just my opinion. Open for others' input.

Comment (allencloud) [inline cli/command/service/scale.go]: WDYT?
Any feedback? @thaJeztah

Comment (thaJeztah) [inline cli/command/service/scale.go]: Oh! Sorry missed this one, yes; perhaps it makes sense

File: integration-cli/docker_cli_service_scale_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	c.Assert(err, checker.IsNil)
+
+	out, err = d.Cmd("service", "scale", "TestService1=2")
+	c.Assert(err, checker.IsNil)
+
+	// TestService2 is a global mode
+	out, err = d.Cmd("service", "scale", "TestService2=2")
+	c.Assert(err, checker.NotNil)
+
+	str := fmt.Sprintf("%s: scale can only be used with replicated mode\n", service2Name)
```
Comment (thaJeztah) [inline integration-cli/docker_cli_service_scale_test.go]: This test doesn't seem to be related to the change made in this PR. Perhaps a test for an invalid scale _value_? (`scale TestService=foobar`, `TestService=-1`)?

Comment (allencloud) [inline integration-cli/docker_cli_service_scale_test.go]: Thanks, Updated.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by allencloud** [comment]
@vdemeester PTAL 🐯

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
I feel some of the integration tests might better be unit test (as for some of them we don't need to start a daemon)
